examples/cms-agilitycms/pages/api/preview.js

```diff
@@ -1,24 +1,35 @@
-import { validatePreview } from '../../lib/api'
-
-export default async function handler(req, res) {
-  // Check the secret and next parameters
-  // This secret should only be known to this API route and the CMS
-
-  //validate our preview key, also validate the requested page to preview exists
-  const validationResp = await validatePreview({
-    agilityPreviewKey: req.query.agilitypreviewkey,
-    slug: req.query.slug,
-    contentID: req.query.contentid,
-  })
-
-  if (validationResp.error) {
-    return res.status(401).end(`${validationResp.message}`)
-  }
-
-  //enable preview mode
-  res.setPreviewData({})
-
-  // Redirect to the slug
-  res.writeHead(307, { Location: validationResp.slug })
-  res.end()
-}
+import { validatePreview, getDynamicPageURL } from '@agility/nextjs/node'
+
+// A simple example for testing it manually from your browser.
+// If this is located at pages/api/preview.js, then
+// open /api/preview from your browser.
+export default async (req, res) => {
+
+	//validate our preview key, also validate the requested page to preview exists
+	const validationResp = await validatePreview({
+		agilityPreviewKey: req.query.agilitypreviewkey,
+		slug: req.query.slug
+	});
+
+	if (validationResp.error) {
+		return res.status(401).end(`${validationResp.message}`)
+	}
+
+	let previewUrl = req.query.slug;
+
+	//TODO: these kinds of dynamic links should work by default (even outside of preview)
```

Comment:
TODO

Reply:
created an issue so we have this todo tracked: https://github.com/agility/agilitycms-nextjs-starter/issues/25